djinni/jni/djinni_jni_main.hpp

```diff
@@ -13,9 +14,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 //
-
-// This provides a minimal JNI_OnLoad and JNI_OnUnload implementation - include it if your
-// app doesn't use JNI except through Djinni.
+// This provides a minimal JNI_OnLoad and JNI_OnUnload implementation.
+// Don't include it if your library does require a custom JNI_OnLoad implementation with custom initialization logic.
+#pragma once
```

Comment:
I've transformed `djinni_main.cpp` to a header file. That way anybody can include it and get the `JNI_OnLoad` implementation. This only works if the header is included once. Otherwise the linker will complain.